### PR TITLE
Add Aurora DNS API

### PR DESCRIPTION
**Pull request by JaZo**
This PR adds support for the Aurora DNS API offered by the Dutch cloud provider PCextreme B.V.

Documentation: https://github.com/acmesh-official/acme.sh/wiki/dnsapi#131-using-the-aurora-api
Issue: https://github.com/acmesh-official/acme.sh/issues/3459
Latest actions run: https://github.com/JaZo/acme.sh/actions/runs/753125822

### CODE REVIEW

File: dnsapi/dns_aurora.sh [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+  if ! _get_root "$fulldomain"; then
+    _err "invalid domain"
+    return 1
+  fi
+  _debug _domain_id "$_domain_id"
+  _debug _sub_domain "$_sub_domain"
+  _debug _domain "$_domain"
+
+  #save the zone id to the account conf file.
+  _saveaccountconf_mutable AURORA_Zone_ID "$_domain_id"
```
Comment (Neilpang) [inline dnsapi/dns_aurora.sh:45]: I don't think you need AURORA_Zone_ID.

The AURORA_Zone_ID is saved in the account conf, so it's global.

what if you have 2 zones there?


If you used AURORA_Zone_ID for ZoneA.com,   you will have problem to issue certs for ZoneB.com.

**Comment by Neilpang** [comment]
BTW, please fix your CI checks:
https://github.com/JaZo/acme.sh/actions/runs/692884497

It's not passing.

File: dnsapi/dns_aurora.sh (hunk 1 above)
Comment (JaZo) [reply dnsapi/dns_aurora.sh]: You're right, I haven't thought about multiple zones/domains. I've completely removed the `AURORA_Zone_ID` variable!

**Comment by JaZo** [comment]
@Neilpang I've re-ran the action several times, but every time it fails at a different point. [The logs](https://github.com/acmesh-official/acme.sh/files/6221009/5_Run.acmetest.txt) also don't give me a clue of what went wrong. One time it got through and even reached the FreeBSD job and then it failed on some [SSH and Rsync issues](https://github.com/acmesh-official/acme.sh/files/6221015/6_FreeBSD.txt). Do you have a direction I should look? Those issues don't seem to be related to my new API implementation, but maybe I'm overlooking something?

**Comment by Neilpang** [comment]
please set a larger TEST_DNS_SLEEP value.

https://github.com/acmesh-official/acme.sh/wiki/DNS-API-Test

**Comment by JaZo** [comment]
@Neilpang My implementation had some issues on OS X, but I managed to fix those. Build is successfull now, so it's ready for review!

N.B. shfmt job is failing, but that's caused by a different DNS API.

**Pull request merged**